autonomi/src/client/config.rs

```diff
@@ -0,0 +1,315 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use ant_evm::EvmNetwork;
+use ant_networking::{ResponseQuorum, RetryStrategy};
+use libp2p::Multiaddr;
+use std::num::NonZero;
+
+/// Configuration for [`crate::Client::init_with_config`].
+///
+/// Use [`ClientConfig::set_client_operation_config`] to set configure how the client performs operations
+/// on the network.
+#[derive(Debug, Clone, Default)]
+pub struct ClientConfig {
+    /// Whether we're expected to connect to a local network.
+    pub local: bool,
+
+    /// List of peers to connect to.
+    ///
+    /// If not provided, the client will use the default bootstrap peers.
+    pub peers: Option<Vec<Multiaddr>>,
+
+    /// EVM network to use for quotations and payments.
+    pub evm_network: EvmNetwork,
+
+    /// Configuration for operations on the client.
+    ///
+    /// This will be shared across all clones of the client and cannot be changed after initialization.
+    pub operation_config: ClientOperationConfig,
+}
+
+/// Configurations for operations on the client.
+///
+/// Default values are used for each type of data, but you can override them here.
+#[derive(Debug, Clone, Default)]
+pub struct ClientOperationConfig {
+    /// Configuration for chunk operations.
+    pub(crate) chunk_operation_config: ChunkOperationConfig,
+    /// Configuration for graph operations.
+    pub(crate) graph_operation_config: GraphOperationConfig,
+    /// Configuration for pointer operations.
+    pub(crate) pointer_operation_config: PointerOperationConfig,
+    /// Configuration for scratchpad operations.
+    pub(crate) scratchpad_operation_config: ScratchpadOperationConfig,
```

Comment:
Do these four really need to be 4 different types?? They look the same to me

Reply:
Yeah, I was thinking about the same and decided on 4 types for these reasons:
- You avoid having to deal with matching if you have a generic `NetworkOperationConfig`, and we also avoid creating a flag/enum to denote each type of data.
  - We could ofc have a generic struct inside each of these `*OperationConfig` to reduce some duplication of the definitions, but since this is a simple config struct, it would be nice if things remained flat?
- Each data type could get it's own set of unique configs later down the road, and this makes things easier.